sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/cloud/core/resource/StorageFileResource.java

```diff
@@ -197,24 +193,16 @@ StorageType getStorageType() {
         return StorageType.FILE;
     }
 
-    private void assertExisted() throws FileNotFoundException {
-        if (!exists()) {
-            throw new FileNotFoundException("File or container not existed.");
-        }
-    }
-
     private void create() throws ShareStorageException {
-        if (!shareClient.exists()) {
-            this.shareClient.create();
-        }
-        if (!shareFileClient.exists()) {
+        if (!this.exists()) {
             ShareFileHttpHeaders header = null;
             if (StringUtils.hasText(contentType)) {
                 header = new ShareFileHttpHeaders();
                 header.setContentType(contentType);
             }
-            //TODO: create method must provide file length, but we don't know actual
-            this.shareFileClient.createWithResponse(1024, header, null, null, null, null, Context.NONE).getValue();
+            // TODO: create method must provide maximum size, but we don't know actual
+            this.shareFileClient.createWithResponse(1024, header, null, null, null, null, Context.NONE)
+                .getValue();
```

Comment:
Is there a way to make `1024` configurable? This is a pretty small number for average application.

Reply:
Thanks @kasobol-msft, we really should not use this `1024` value here. But in a `Resource` implementation, it's not reasonable for users to define a `maximum size` for a file (if the file does not exist). The users locate a resource using `azure-file://some-share/some-file`. 

So we have to figure out how to make it configurable. How about implementing this feature request in another PR?

Reply:
sounds good. please create GH issue to track.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/30120